fem/quadinterpolator.hpp

```diff
@@ -101,41 +116,61 @@ class QuadratureInterpolator
    /// Interpolate the values of the E-vector @a e_vec at quadrature points.
    void Values(const Vector &e_vec, Vector &q_val) const;
 
-   /** @brief Interpolate the derivatives of the E-vector @a e_vec at quadrature
-       points. */
+   /** @brief Interpolate the derivatives (with respect to reference
+       coordinates) of the E-vector @a e_vec at quadrature points. */
    void Derivatives(const Vector &e_vec, Vector &q_der) const;
 
    /** @brief Interpolate the derivatives in physical space of the E-vector
        @a e_vec at quadrature points. */
    void PhysDerivatives(const Vector &e_vec, Vector &q_der) const;
 
+   /** @brief Compute the determinants of the derivatives (with respect to
+       reference coordinates) of the E-vector @a e_vec at quadrature points. */
+   void Determinants(const Vector &e_vec, Vector &q_det) const;
+
    /// Perform the transpose operation of Mult(). (TODO)
    void MultTranspose(unsigned eval_flags, const Vector &q_val,
                       const Vector &q_der, Vector &e_vec) const;
 
    // Compute kernels follow (cannot be private or protected with nvcc)
```

Comment:
I'm thinking, it's probably better to hide these compute kernels as static functions in the `.cpp` file. Any objections?

Reply:
No objections

Reply:
These kernels are in other `cpp` files to avoid having too many instanciations in a same one.
If we have an elegant way to keep this but hide them, sure.

Reply:
I pushed an update that removes these template member functions -- let me know if you have any objections.

Reply:
Looks good!